fsm.go

```diff
@@ -365,3 +383,14 @@ func (s *stateMachine) failedMessage(id id) *message {
 		Addr: s.members[id].addr,
 	}
 }
+
+// addUserMsg adds a new userMsg carrying b to the user message queue.
+func (s *stateMachine) addUserMsg(b []byte) {
+	messageID := randID()
+	s.userMsgQueue.Upsert(messageID, &message{
+		Type:      userMsg,
+		ID:        s.id,
+		MessageID: messageID,
```

Comment:
It's definitely kinda confusing to have a field called `message.ID` and another field `message.MessageID`, when neither is the ID of the `message` (though one is the ID of the `Message`).

Ideas:

- maybe field `ID` should be `NodeID`
- maybe `Message` should be called something else
- maybe `message` should be called something else